core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3599,8 +3600,8 @@ private boolean isSingleGatheringExchange(PlanNode node)
                     .findFirst();
 
             return result.isPresent()
-                    && result.get() instanceof ExchangeNode
-                    && ((ExchangeNode) result.get()).getPartitioningScheme().getPartitioning().getHandle().equals(SINGLE_DISTRIBUTION);
+                    && result.get() instanceof ExchangeNode exchangeNode
```

Comment:
Are the changes in java files necessary ?

Reply:
Yes. These are error prone related